tests/test_ramp_fitting.py

```diff
@@ -452,6 +452,241 @@ def test_2_group_cases():
     np.testing.assert_allclose(err, chk_er, tol)
 
 
+def run_one_group_ramp_suppression(nints, suppress):
+    """
+    Forms the base of the one group suppression tests.  Create three ramps
+    using three pixels with two integrations.  In the first integration:
+        The first ramp has no good groups.
+        The second ramp has one good groups.
+        The third ramp has all good groups.
+
+    In the second integration all pixels have all good groups.
+    """
+    # Define the data.
+    ngroups, nrows, ncols = 5, 1, 3
+    dims = (nints, ngroups, nrows, ncols)
+    rnoise, gain = 10, 1
+    nframes, group_time, frame_time = 1, 5.0, 1
+    var = rnoise, gain
+    tm = nframes, group_time, frame_time
+
+    # Using the above create the classes and arrays.
+    ramp_data, rnoise2d, gain2d = setup_inputs(dims, var, tm)
+
+    arr = np.array([k + 1 for k in range(ngroups)], dtype=float)
+
+    sat = ramp_data.flags_saturated
+    sat_dq = np.array([sat] * ngroups, dtype=ramp_data.groupdq.dtype)
+    zdq = np.array([0] * ngroups, dtype=ramp_data.groupdq.dtype)
+
+    ramp_data.data[0, :, 0, 0] = arr
+    ramp_data.data[0, :, 0, 1] = arr
+    ramp_data.data[0, :, 0, 2] = arr
+
+    ramp_data.groupdq[0, :, 0, 0] = sat_dq  # All groups sat
+    ramp_data.groupdq[0, :, 0, 1] = sat_dq  # 0th good, all others sat
```

Comment:
I think you need `ramp_data.groupdq[0, 1:, 0, 1] = sat_dq` here, so that everything *but* the 0th group is flagged.

Reply:
The next line sets the zeroeth group to zero for that ramp.